### PR TITLE
chore: add error handling for bad timers and deadlines when making seats

### LINKED ISSUE #1538: [Zoe] errors are silently lost in setWakeup
The promise result is not wired in anywhere so errors are silently lost. What happens if the setWakeup doesn't work? What is the time is already past?

the promise result should be tied into errors for this offer.

_Originally posted by @dtribble in https://github.com/_render_node/MDExOlB1bGxSZXF1ZXN0NDY1NjU3MTI5/timeline/more_items_

### PR DESCRIPTION
**Pull request by katelynsills**
Add error handling when `setWakeup` throws. Add a better error message than whatever caused the attempt to fail.

Closes #1538

### CODE REVIEW

**Review by Chris-Hibbert — CHANGES_REQUESTED** [review]
Please add a test that shows the new behavior.

**Comment by katelynsills** [comment]
> Please add a test that shows the new behavior.

Sure thing

File: packages/zoe/src/contractFacet/exit.js [hunk 1]
```diff
@@ -26,12 +26,21 @@ export const makeExitObj = (proposal, zoeSeatAdmin, zcfSeatAdmin) => {
...
+        harden({
+          wake: exitFn,
+        }),
+      )
+      .catch(err => {
+        console.error(
+          `The seat could not be made with the provided timer ${proposal.exit.afterDeadline.timer} and deadline ${proposal.exit.afterDeadline.deadline}`,
+        );
+        console.error(err);
+        exitFn();
```
Comment (erights) [inline packages/zoe/src/contractFacet/exit.js]: `exitFn` does an `exit`. Here, I think you want to do a `kickOut`.

**Review by erights — CHANGES_REQUESTED** [review]

File: packages/zoe/src/contractFacet/exit.js (hunk 1 above)
Comment (katelynsills) [reply packages/zoe/src/contractFacet/exit.js]: Ah, very good point! I will fix that

File: packages/zoe/src/contractFacet/exit.js [hunk 2]
```diff
@@ -26,12 +26,22 @@ export const makeExitObj = (proposal, zoeSeatAdmin, zcfSeatAdmin) => {
...
-      }),
-    );
+    E(proposal.exit.afterDeadline.timer)
+      .setWakeup(
+        proposal.exit.afterDeadline.deadline,
+        harden({
+          wake: exitFn,
+        }),
+      )
+      .catch(err => {
```
Comment (erights) [inline packages/zoe/src/contractFacet/exit.js]: For errors that manifest as a rejected promise's rejection reason, prefer `reason` to `err`, but either is acceptable.

**Review by erights — APPROVED** [review]
LGTM

File: packages/zoe/src/contractFacet/exit.js (hunk 2 above)
Comment (katelynsills) [reply packages/zoe/src/contractFacet/exit.js]: Ok, sure, that sounds good. I'll change it.

**Pull request merged**